langserver/definition.go

```diff
@@ -108,14 +133,36 @@ func (h *LangHandler) handleXDefinition(ctx context.Context, conn jsonrpc2.JSONR
 		return nil, err
 	}
 
-	var nodes []*ast.Ident
+	var nodes []foundNode
 	obj, ok := pkg.Uses[node]
 	if !ok {
 		obj, ok = pkg.Defs[node]
 	}
 	if ok && obj != nil {
 		if p := obj.Pos(); p.IsValid() {
-			nodes = append(nodes, &ast.Ident{NamePos: p, Name: obj.Name()})
+			typ := pkg.TypeOf(node).String()
+			typIdent := typ
+			var typObj types.Object
+			if idx := strings.LastIndex(typ, "."); idx != -1 {
```

Comment:
I don't think this makes sense for any composite types. See the implementation of how these strings are created and this would likely break structs, maps, etc. https://sourcegraph.com/github.com/golang/go@go1.10/-/blob/src/go/types/typestring.go#L78

You probably need some sort of solution which does the correct thing for each possible type.